highfidelity/hifi
# Edit handle scale improvements

Replace the proportional scaling cubes at the corners of the bounding box with a single scale cube in the center of the entity. This cube behaves like the scale cube in Unity in that rightward or upward mouse movement will scale outward, while leftward or downward mouse movement will scale inward. As well, once scaling downward passed the minimum dimensions, it will invert and start scaling back upward.

Replace the single axis stretching spheres with cubes.

Maintain stretch cube positioning relative to mouse when stretching, then revert back to default handle positioning when done stretching.

Addresses:
https://highfidelity.manuscript.com/f/cases/17467
https://highfidelity.manuscript.com/f/cases/17474
https://highfidelity.manuscript.com/f/cases/17470

Test Plan:
- Enter Interface in desktop mode
- In Create app, create any entity and select it or select any existing entity
- Verify the edit handles to stretch each axis are now cubes instead of spheres
- Verify that when dragging a stretch cube on each axis that the stretch cube stays under the mouse's position until it is released, at which point it goes back to it's default position aligned with the other handles
- Verify the edit handle cubes at the corners of the bounding box are now gone
- Verify the new edit handle cube at the center of the entity performs proportional scaling when dragged:
-- rightward or upward mouse movement proportionally scales the entity upward
-- leftward or downward mouse movement will scale it downward
-- scaling downward passed minimum dimensions will invert and start scaling upward again
- Verify the center scale cube proportionally scales entities with dimensions of (100,1,1), (1,100,1), and (1,1,100) all at the same rate
